File: pysrc/send.py
#!/usr/bin/env python3

import os, sys, socket

if len(sys.argv) < 2:
    print("Usage: python send.py <chunk-file>")
    sys.exit(1)

chunk_file = sys.argv[1]

HOST = os.getenv("HOST", "127.0.0.1")
PORT = int(os.getenv("RECV_PORT", "6000"))

s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.connect((HOST, PORT))

with open(chunk_file, "rb") as f:
    chunk = f.read()

s.sendall(b"python:" + chunk)
s.close()

File: luasrc/send.lua
#!/usr/bin/env lua

local socket = require("socket")

if #arg < 1 then
    print("Usage: lua send.lua <chunk-file>")
    os.exit(1)
end

local chunk_file = arg[1]

local host = os.getenv("HOST") or "127.0.0.1"
local port = tonumber(os.getenv("RECV_PORT") or "6000")

local client = assert(socket.tcp())
assert(client:connect(host, port))

local f = assert(io.open(chunk_file, "rb"))
local chunk = f:read("*all")
f:close()
local msg = "lua:" .. chunk

client:send(msg)
client:close()
